src/cdk/tree/tree.ts

```diff
@@ -570,6 +679,39 @@ export class CdkTree<T, K = T> implements AfterContentChecked, CollectionViewer,
     // - if it's not, then K will be defaulted to T.
     return this.expansionKey?.(dataNode) ?? (dataNode as unknown as K);
   }
+
+  /**
+   * Converts children for certain tree configurations. Note also that this
+   * caches the known nodes for use in other parts of the tree.
+   */
+  private _convertChildren(nodes: readonly T[]): Observable<readonly T[]> {
+    // The only situations where we have to convert children types is when
+    // they're mismatched; i.e. if the tree is using a childrenAccessor and the
+    // nodes are flat, or if the tree is using a levelAccessor and the nodes are
+    // nested.
```

Comment:
I don't quite follow how this situation would come about- I would think we would only want to support _either_ flat or nested for the entire free instance